New Checkout Pro documentation (#1301)
mercadopago/devsite-docs: guides/online-payments/checkout-pro/advanced-integration.pt.md

# Integração avançada

> Otimize a sua integração com funcionalidades extras que te permitirão melhor gerenciar os pagamentos de suas vendas com o Checkout Pro.

Reviewer: ```suggestion
O **Checkout Pro** dispõe de funcionalidades extras que permitem otimizar sua integração e melhorar a gestão dos pagamentos de suas vendas.
```

## Receba notificações de pagamentos

> As notificações IPN (_Instant Payment Notification_) são uma forma automática de aviso da criação de novos pagamentos e as atualizações de seus _status_, isto é, se eles foram aprovados, recusados ou se estão pendentes.

Reviewer: ```suggestion
As notificações IPN (_Instant Payment Notification_) são notificações automáticas sobre a criação de novos pagamentos e atualizações de status, informando se as transações foram aprovadas, recusadas ou estão pendentes.
```

> Os avisos automáticas permitem que você administre seu estoque e mantenha seu sistema sincronizado com os fluxos de pagamento do seu negócio. Aprenda como receber as notificações IPN [aqui](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/notifications/ipn)

Reviewer: ```suggestion
Os avisos automáticos permitem que você administre seu estoque e mantenha seu sistema sincronizado com os fluxos de pagamento do seu negócio. Aprenda como receber as notificações IPN [aqui](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/notifications/ipn)
```

> ## Adicione informações adicionais à preferência

Reviewer: ```suggestion
## Insira informações adicionais à preferência
```

> Melhore a aprovação dos pagamentos e a experiência dos seus compradores no Checkout Pro adicionando informações às suas preferências que possam detalhar sobre o item comprado e o usuário comprador.

Reviewer: ```suggestion
Melhore a aprovação dos pagamentos e a experiência dos seus compradores no Checkout Pro adicionando informações às suas preferências que possam detalhar sobre o item comprado e o usuário comprador.
```
```suggestion
Melhore a aprovação dos pagamentos e a experiência de compra no Checkout Pro adicionando informações às suas preferências que detalhem o item comprado e o usuário comprador.
```

> As informações adicionais passíveis de serem adicionadas por você nas preferências são:

Reviewer: ```suggestion
Em Preferências, você pode inserir as seguintes informações adicionais:
```

```
]]]

> NOTE
> 
> Nota
> 
> Você pode encontrar a lista de categorias do seu `item` [aqui](https://api.mercadopago.com/item_categories). Se não conseguir acessar os valores de categoria, envie o valor `others` no atributo `category_id`.

## Redirecione o comprador para o seu _site*_

No final do processo de pagamento, você tem a opção de redirecionar o comprador para o seu _site_ novamente. 

Para isso, adicione o atributo `back_urls` e defina, segundo o status do pagamento, a página desejada para redirecionar o seu comprador quando ele clicar no botão de retorno ao site.

> Se quer que o redirecionamento seja automático para os pagamentos aprovados, sem a renderização de um botão de retorno, é preciso adicionar também o atributo `auto_return` com valor `approved`. 

Reviewer: ```suggestion
Se deseja que o redirecionamento para os pagamentos aprovados seja automático, sem a renderização de um botão de retorno, é preciso adicionar também o atributo `auto_return` com valor `approved`. 
```

> NOTE
>
> Nota
>
> Tenha em conta que o atributo `auto_return` só funciona para o modo `redirect` e `mobile` do Checkout Pro. O mesmo não funciona no [modo modal](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/online-payments/checkout-pro/integration), já que nesse último o comprador segue no _site_ durante todo o processo de pagamento.

![autoreturn](/images/web-payment-checkout/autoreturn-img-br.png)

| Atributo |	Descrição |
| ------------ 	|	-------- | 

> | `auto_return` | Redireciona automaticamente compradores para o _site_ quando o pagamento é finalizado como aprovado. O valor padrão é `approved`. |

Reviewer: ```suggestion
| `auto_return` | Os compradores são redirecionados automaticamente para o  _site_ quando o pagamento é aprovado. O valor padrão é `approved`. |
```

| `back_urls` | URL de retorno ao site. Possíveis cenários são: `success`: URL de retorno perante pagamento aprovado.<br/><br/>`pending`: URL de retorno perante pagamento pendente.<br/><br/>`failure`: URL de retorno perante pagamento rejeitado.

Através das `back_urls`, serão retornados os seguintes parâmetros:

| Parâmetro |	Descrição |

> | ------------ 	|	-------- | 

Reviewer: ```suggestion
| --- | --- | 
```

## Evite recusas de pagamentos

Um pagamento pode ser recusado porque o emissor do meio de pagamento detectou um problema no fluxo ou porque o mesmo não preencheu os requisitos de segurança necessários.

Evite pagamentos recusados com as nossas boas práticas e [melhore a taxa de aprovação dos seus pagamentos](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/manage-account/account/payment-rejections).

## Gerencie cancelamentos e estornos

----[mla, mlm, mco, mlu, mlb, mlc]----
Os cancelamentos são feitos quando o pagamento não foi concluído antes da data de vencimento e o vendedor decide então cancelá-lo. As devoluções, por sua vez, acontecem quando o pagamento foi feito pelo comprador, mas o vendedor decide estorná-lo, total ou parcialmente.
------------

----[mpe]----
Os cancelamentos são feitos quando o pagamento não foi concluído antes da data de vencimento e o vendedor decide então cancelá-lo. As devoluções, por sua vez, acontecem quando o pagamento foi feito pelo comprador, mas o vendedor decide estorná-lo totalmente.
------------

> Encontre todas as informações necessárias para melhor gerenciar os cancelamentos e estornos dos seus pagamentos na documentação [Devoluções e cancelamentos](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/manage-account/account/cancellations-and-refunds) do Mercado Pago.

Reviewer: ```suggestion
Para mais informações sobre como gerenciar os cancelamentos e estornos dos seus pagamentos, acesse a documentação [Devoluções e cancelamentos](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/manage-account/account/cancellations-and-refunds) do Mercado Pago.
```

## Gerencie contestações

> Uma contestação (ou *chargeback*) acontece quando o comprador entra em contato com a entidade emissora do cartão e desconhece o pagamento. Na prática, isso significa que o dinheiro desse pagamento será retido da sua conta Mercado Pago até que seja a situação seja solucionada.

Reviewer: ```suggestion
Uma contestação (ou *chargeback*) acontece quando o comprador entra em contato com a entidade emissora do cartão e desconhece o pagamento. Na prática, isso significa que o dinheiro desse pagamento será retido da sua conta Mercado Pago até que a situação seja solucionada.
```
